snapcraft/internal/sources.py

```diff
@@ -201,6 +201,39 @@ def pull(self):
         subprocess.check_call(cmd)
 
 
+class Subversion(Base):
+
+    def __init__(self, source, source_dir, source_tag=None,
+                 source_branch=None):
+        super().__init__(source, source_dir, source_tag, source_branch)
+        if source_tag:
+            if source_branch:
+                raise IncompatibleOptionsError(
+                    "Can't specify source-tag OR source-branch for a "
+                    "Subversion source")
+            else:
+                raise IncompatibleOptionsError(
+                    "Can't specify source-tag for a Subversion source")
+        elif source_branch:
+            raise IncompatibleOptionsError(
+                "Can't specify source-branch for a Subversion source")
+
+    def pull(self):
+        if os.path.exists(os.path.join(self.source_dir, '.svn')):
+            subprocess.check_call(
+                ['svn', 'update'], cwd=self.source_dir)
+        else:
+            if self.source.startswith('svn:') or \
+                    self.source.startswith('https:') or \
+                    self.source.startswith('http:') or \
+                    self.source.startswith('svn+ssh:'):
```

Comment:
I'm wondering what would be a better way to do this.
what about reversing the loging and
if os.path.isdir(self.source), append file://, else, leave it alone ?